### DISCUSSION TITLE
Access via TypeScript

### DISCUSSION THREAD

**Discussion by hamzaelmaghari** [saadeghi/daisyui#1199] [Q&A]
## Is it possible to access active theme colors from JavaScript or TypeScript file ?

**Comment by ichenglin** [comment]
In **tailwind and daisyui**, you could **access the active configuration** with the **[`resolveConfig()`](https://tailwindcss.com/docs/configuration#referencing-in-java-script)** method.
It allows **Javascript** (or **Typescript** in this case) to get information such as the **active theme color**.

> **Referencing in JavaScript [(documentation)](https://tailwindcss.com/docs/configuration#referencing-in-java-script)**
>
> It can often be useful to reference your configuration values in your own client-side JavaScript — for example to access some of your theme values when dynamically applying inline styles in a React or Vue component.
> To make this easy, Tailwind provides a resolveConfig helper you can use to generate a fully merged version of your configuration object.

Here's an **example** of using the **[`resolveConfig()`](https://tailwindcss.com/docs/configuration#referencing-in-java-script)** method. *(copied from documentation)*
```js
import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from './tailwind.config.js'

const fullConfig = resolveConfig(tailwindConfig)

fullConfig.theme.width[4]
// => '1rem'

fullConfig.theme.screens.md
// => '768px'

fullConfig.theme.boxShadow['2xl']
// => '0 25px 50px -12px rgba(0, 0, 0, 0.25)'
```

**Reply by hamzaelmaghari** [reply]
Unfortunately, this didn't fit my needs I need to get the active `daisyui` active theme primary color since my application has a theme changer.

As we can see here this method showed up can access only static variables located in `tailwind.config.js`.

**Comment by wuarmin** [comment]
Checkout #415
